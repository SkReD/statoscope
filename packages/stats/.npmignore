*
!bin
!spec/**/*.{js,d.ts,json}
**/*.spec.{js,d.ts}
