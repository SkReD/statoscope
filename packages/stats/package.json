{
  "name": "@statoscope/stats",
  "version": "5.28.1",
  "description": "Statoscope stats spec",
  "types": "./spec/index.d.ts",
  "repository": "https://github.com/statoscope/statoscope/",
  "author": {
    "name": "Sergey Melyukov",
    "url": "https://twitter.com/smelukov"
  },
  "publishConfig": {
    "access": "public"
  },
  "files": [
    "spec/*"
  ],
  "license": "MIT",
  "homepage": "https://github.com/statoscope/statoscope/"
}
