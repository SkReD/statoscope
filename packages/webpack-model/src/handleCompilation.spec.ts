import normalize from './handleFile';
import stats from '../../../test/bundles/v5/simple/stats-prod.json';
const normalized = normalize({ name: 'stats.js', data: stats });
const firstFile = normalized.files[0];
const compilation = firstFile.compilations[0];

it('extractModulePackages', () => {
  expect(
    compilation.nodeModules.map((nm) => ({
      name: nm.name,
      instances: nm.instances.map((i) => ({
        modules: i.modules.map((m) => m.resolvedResource),
        reasons: i.reasons.map((r) => ({
          type: r.type,
          data: r.data.resolvedResource,
        })),
      })),
    })),
  ).toMatchSnapshot();
});
