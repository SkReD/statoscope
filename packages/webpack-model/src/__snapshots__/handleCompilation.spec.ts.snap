// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`extractModulePackages 1`] = `
[
  {
    "instances": [
      {
        "modules": [
          "node_modules/is-array/index.js",
        ],
        "reasons": [
          {
            "data": "simple/src/index.ts",
            "type": "module",
          },
          {
            "data": "simple/src/index2.ts",
            "type": "module",
          },
        ],
      },
    ],
    "name": "is-array",
  },
  {
    "instances": [
      {
        "modules": [
          "simple/node_modules/foo/index.js",
        ],
        "reasons": [
          {
            "data": "simple/src/sub-module.ts",
            "type": "module",
          },
        ],
      },
    ],
    "name": "foo",
  },
  {
    "instances": [
      {
        "modules": [
          "simple/node_modules/bar/index.js",
        ],
        "reasons": [
          {
            "data": "simple/src/sub-module.ts",
            "type": "module",
          },
        ],
      },
      {
        "modules": [
          "simple/node_modules/foo/node_modules/bar/index.js",
        ],
        "reasons": [
          {
            "data": "simple/src/index.ts",
            "type": "module",
          },
          {
            "data": "simple/node_modules/foo/index.js",
            "type": "module",
          },
        ],
      },
    ],
    "name": "bar",
  },
  {
    "instances": [
      {
        "modules": [
          "node_modules/css-loader/dist/runtime/api.js",
          "node_modules/css-loader/dist/runtime/getUrl.js",
        ],
        "reasons": [
          {
            "data": "simple/src/index2.css",
            "type": "module",
          },
        ],
      },
    ],
    "name": "css-loader",
  },
]
`;
